src/index.ts

```diff
@@ -60,58 +65,63 @@ async function run(): Promise<void> {
         }
         else {
             core.warning(`The version of ${imageToPush} in the Podman image storage is more recent `
-                + `than the version in the Docker image storage. The image from the Podman image `
+                + `than the version in the Docker image storage. Tag(s) of the image from the Podman image `
                 + `storage will be pushed.`);
         }
     }
     else if (isPresentInDocker) {
         imageToPush = `${dockerBaseUrl}/${imageToPush}`;
         core.info(`${imageToPush} was found in the Docker image storage, but not in the Podman `
-            + `image storage. The image will be pulled into Podman image storage, pushed, and then `
+            + `image storage. Tag(s) of the image will be pulled into Podman image storage, pushed, and then `
             + `removed from the Podman image storage.`);
         isImageFromDocker = true;
     }
 
-    let pushMsg = `Pushing ${imageToPush} to ${registry}`;
+    let pushMsg = `Pushing ${imageToPush} with tags ${tagsList.toString()} to ${registry}`;
     if (username) {
         pushMsg += ` as ${username}`;
     }
     core.info(pushMsg);
 
     const registryWithoutTrailingSlash = registry.replace(/\/$/, "");
-    const registryPath = `${registryWithoutTrailingSlash}/${imageInput}:${tag}`;
 
     const creds = `${username}:${password}`;
 
     let digestFile = digestFileInput;
+    const imageNameWithTag = `${imageToPush}:${tagsList[0]}`;
     if (!digestFile) {
-        digestFile = `${imageToPush.replace(
+        digestFile = `${imageNameWithTag.replace(
             /[/\\/?%*:|"<>]/g,
             "-",
         )}_digest.txt`;
     }
 
     // push the image
-    const args = [
-        "push",
-        "--quiet",
-        "--digestfile",
-        digestFile,
-        "--creds",
-        creds,
-        imageToPush,
-        registryPath,
-    ];
-
-    // check if tls-verify is not set to null
-    if (tlsVerify) {
-        args.push(`--tls-verify=${tlsVerify}`);
-    }
+    for (const tag of tagsList) {
+        const imageWithTag = `${imageToPush}:${tag}`;
+        const registryPath = `${registryWithoutTrailingSlash}/${imageInput}:${tag}`;
```

Comment:
why not just `${registryWithoutTrailingSlash}/${imageWithTag}`;

Reply:
Actually `imageWithTag` may contain `docker.io/library/myimage:v1` for the image pulled from docker storage and we don't want to have `docker.io/library` in registry path

Reply:
oh, ok. it was hard to track the value of `imageInput` because it is a `let` defined so far up.

Reply:
Yeah